src/ast.rs

```diff
@@ -549,11 +555,72 @@ impl Module {
 // Virtual modules cannot be compile targets
 // Please see doc/modules.md for more info
 impl Virtualable for Module {
-    fn set_virtual(&mut self) {
+    fn set_virtual(&mut self, _range: Range<usize>) -> Result<(), ParseErrorMsg> {
         self.is_virtual = true;
+        Ok(())
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct System {
+    pub name: CascadeString,
+    pub annotations: Annotations,
+    pub modules: Vec<CascadeString>,
+    pub configurations: Vec<LetBinding>,
+}
+
+impl System {
+    pub fn new(name: CascadeString) -> Self {
+        System {
+            name,
+            annotations: Annotations::new(),
+            modules: Vec::new(),
+            configurations: Vec::new(),
+        }
+    }
+
+    pub fn set_fields(mut self, input: Vec<SystemBody>) -> Self {
+        for i in input {
+            match i {
+                SystemBody::Mod(m) => {
+                    self.modules.push(m);
+                }
+                SystemBody::Config(l) => {
+                    self.configurations.push(l);
+                }
+            }
+        }
+
+        // Insert the default configurations if they were not provided
+        for (config_name, default_value) in constants::SYSTEM_CONFIG_DEFAULTS {
+            let config_name = config_name.to_string();
```

Comment:
Is this conversion actually necessary?  I believe that PartialEq is defined on &str and String and that From<&str> is defined for CascadeString.  So I'd think that we could just use a &str everywhere in this loop without converting to String first?